core/src/main/java/com/linecorp/armeria/server/TransientService.java

```diff
@@ -31,6 +36,17 @@
 @FunctionalInterface
 public interface TransientService<I extends Request, O extends Response> extends Service<I, O> {
 
+    /**
+     * Returns a new {@link HttpService} decorator which makes the specified {@link HttpService} as
+     * {@link TransientService}.
+     */
+    static Function<? super HttpService, SimpleDecoratingHttpService> newDecorator(
+            TransientServiceOption... transientServiceOptions) {
+        requireNonNull(transientServiceOptions, "transientServiceOptions");
+        return delegate -> new WrappingTransientHttpService(delegate,
+                                                            ImmutableSet.copyOf(transientServiceOptions));
```

Comment:
- Could use `EnumSet` such as `Sets.immutableEnumSet()`?
- Add another factory method for `Iterable<TransientServiceOption> transientServiceOptions`?